### DISCUSSION TITLE
Replacement for old (black) Sonoff RF Bridge

### DISCUSSION THREAD

**Discussion by HaViGit** [arendst/Tasmota#16666] [General]
Can someone tell me if there are alternatives to the black Sonoff RF Bridge? Tasmota works fine on my old (black) Sonoff WiFi Bridge, but it gives problems on the new (white) one. That's why I'm looking for alternative hardware on which I can run Tasmota.

**Comment by sfromis** [comment]
Looking for "RF" has 3 hits on this repository page: https://templates.blakadder.com/misc.html

Additionally, as Tasmota can handle RF without a dedicated extra chip, an option is to connect transmitter/receiver boards to a dev board. https://tasmota.github.io/docs/RF-Transceiver/

**Reply by HaViGit** [reply]
Thanks for your comment.

When I search for RF on the page you mention I come across two bridges, namely the discontinued Sonoff and a VirageBridge RF Gateway that as far as I can see is not for sale in the EU. Considering the price (plus postage and taxes), that doesn't seem like a good alternative to me.

I worked with RF transmitter and receiver modules in combination with Raspberry and Arduino. Unfortunately on the page you refer to (https://tasmota.github.io/docs/RF-Transceiver/) I find very little or no information on how to build something like this with, for example, an ESP-32 board. By the way, this doesn't seem like an MQTT gateway.

Does anyone have experience with this or a link to a project page?

**Reply by sfromis** [reply]
There are 3 devices on the page mentioning RF, not only 2, the third is placed outside of the separate section.

The build for Tasmota should be basically the same as with Raspberry and Arduino, connect power, connect data, and config the pins. This wiring information is already in the linked page, meaning that you have all you need for a build.

ESP32 vs ESP8266 makes no difference for a build, but I don't know if someone has tested such a build with Tasmota on ESP32. With Tasmota based on MQTT, received data is automatically relayed to MQTT, and data can be sent via commands.

**Reply by HaViGit** [reply]
Thanks again! I just flashed Tasmota on a Wemos D1 Mini (ESP8266). Maybe you can tell me what to choose under Module Parameters / Module Type.

Tomorrow I will test it and post the results here.

**Reply by sfromis** [reply]
That specific info about pins is already detailed on https://tasmota.github.io/docs/RF-Transceiver/
You need to upgrade to tasmota-sensors.bin.gz for the RF pin configs to be available in the dropdowns.

You need no change in default module selection for RF, most of the time support for features included in the build can be enabled simply by configuring the appropriate pins.

**Reply by HaViGit** [reply]
After upgrading with tasmota-sensors.bin.gz as suggested by you it worked, i.e. I receive the RF signals. I haven't tested transmitting because I don't need it. I connected the RF transmitter and receiver to the TX and RX pins on the Wemos D1 Mini (_GPIO1_ and _GPIO3_) and chose _Generic(0)_ as the Tasmota module. I power the modules with 5 volts, not 3.3 volt.

Thanks for your help, I couldn't have done it without it!
